python/TestHarness/unit_tests/test_HarnessTester.py

```diff
@@ -95,3 +95,41 @@ def testDuplicateOutputsOK(self):
         self.assertNotRegexpMatches(output, 'heavy_out.e')
         # all
         self.assertNotRegexpMatches(output, 'FATAL TEST HARNESS ERROR')
+
+    def testDeleted(self):
+        """
+        Test that deleted tests returns a failed deleted test when extra info argument is supplied
+        """
+        with self.assertRaises(subprocess.CalledProcessError) as cm:
+            self.runTests('-i', 'deleted', '-e')
+
+        e = cm.exception
+        self.assertRegexpMatches(e.output, 'test_harness\.deleted.*?deleted \(test deleted test\)')
+
+
+    def testDryRun(self):
+        """
+        Test that --dry-run returns a passing 'DRY RUN' status
+        """
+        output = self.runTests('-i', 'diffs', '--dry-run')
+
+        self.assertRegexpMatches(output, 'test_harness\.exodiff.*?DRY RUN')
+        self.assertRegexpMatches(output, 'test_harness\.csvdiff.*?DRY RUN')
+
+        ## TODO: are the following caveats valid when using --dry-run?
+        ## TODO: should we test for 'DRY RUN' with every possible test above as well?
+        ##       for example: failing cyclic tests should _pass_ when using --dry-run
+
+        ## Create caveat tests only for tests that appear to override the 'DRY RUN' response
+        # Skipped caveat test which returns skipped instead of 'DRY RUN'
+        output = self.runTests('-i', 'depend_skip_tests', '--dry-run')
+
+        self.assertIn('skipped (always skipped)', output)
+        self.assertIn('skipped (skipped dependency)', output)
+
+        # Deleted caveat test which returns deleted instead of 'DRY RUN'
+        with self.assertRaises(subprocess.CalledProcessError) as cm:
+            self.runTests('-i', 'deleted', '-e',  '--dry-run')
```

Comment:
I suppose I can perform the dryrun check during the os.walk instead, and return directly to handleResults, thus never allowing checkRunnable to function. But if I do that, we will be informing the user that every test was about to run had they not used the --dry-run argument.

I felt that dry run was not intended to be used in that manner, so I am allowing checkRunnable to perform its normal duties of when a test is allowed to run or not.

Deleted tests are the only test that checkRunnable sets to a status other than skipped (and as such, treated as failures later on). I suppose I can write a small routine specific for that check:

```python
if test == deleted and options.dry_run and options.extra_info:
  do the right thing
```

I just felt, that it was cleaner to allow checkRunnable to run unmolested, and handle the odd case where the user was using --dry-run along with extra info ( -e ). Hence the assertRaises unit test.